libtransmission/peer-mgr.cc

```diff
@@ -857,11 +860,6 @@ class tr_swarm
             // info_that will be replaced by info_this later, so decrement stat
             --stats.known_peer_from_count[info_that->from_first()];
         }
-        // we are going to insert a brand-new peer info object to the pool
-        else if (std::empty(info_this->listen_port()))
-        {
```

Comment:
so I _think_ you are trying to change and make consistent the use of set_connectable(), right?  So that it nows only records successful connections.

The downside to this is that we want a peer that just gave us its listen_port to be relatively highly prioritized.  Every peer in connectable_pool is supposed to have a valid listen_port, but what happens to this peer (in saving resume, on PEX, etc) now that we have it's listen_port but we haven't marked it as connectable?  We will basically never mark it as connectable if we are also a seed, since we'll never start an outbound connection.  So inbound will never get "promoted" to connectable?

Reply:
When saving resume, whether the peer has been marked as connectable or not has no effect.

When sending pex, all it will do is tell the recipient that we aren't sure whether this peer is connectable.
P.S. While on the topic, I am rethinking whether it was a good idea to filter out peers that weren't marked as connectable in `tr_peerMgrAddPex()`. If the peer turned out to be not very useful, it will (or is supposed to) be dropped by the peer info pulse anyway, so why risk missing out on some good but untested peers?

If both the peer and us are seeds, and especially in a private torrent setting, I think it's not worth the cost of establishing a connection just to test if the peer is connectable. In a public torrent, the most we will get out of it is exchanging some peers via pex. In a private torrent, there is virtually no value to trying the connection.

> we want a peer that just gave us its listen_port to be relatively highly prioritized.

What's the reason for it?

Reply:
> In a private torrent, there is virtually no value to trying the connection.

That's exactly what I'm hoping to optimize for---frankly all my testing lately in focused on reducing peer chatter on private torrents, without losing rapid peer discovery on new torrents.

Reply:
I'm confused... So you were disputing about whether to set the connectable flag when we get a port message, right? And somehow setting the connectable flag here would reducing peer chatter? I'm failing to see how that works.

Maybe I should have structured my last reply better..? What I was trying to say is that I think it's fine to leave the `is_connectable_` empty (a.k.a. `is_connectable_.has_value() == false`). Or in other words, I don't think this is an issue:

> We will basically never mark it as connectable if we are also a seed, since we'll never start an outbound connection. So inbound will never get "promoted" to connectable?